src/main/java/org/example/dao/PointDAOImpl.java

```diff
@@ -0,0 +1,104 @@
+package org.example.dao;
+
+import org.example.configurations.ConnectionPool;
+import org.example.model.Point;
+
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class PointDAOImpl implements PointDAO {
+    private static final ConnectionPool CONNECTIONPOOL = ConnectionPool.getInstance();
+
+    public PointDAOImpl() {
+    }
+
+    @Override
+    public void insertPoint(Point point) {
+        Connection connection = CONNECTIONPOOL.getConnectionFromPool();
+        try {
+            PreparedStatement statement = connection.prepareStatement("INSERT INTO point (id, xCoordinate, yCoordinate) VALUES (?, ?, ?)", Statement.RETURN_GENERATED_KEYS);
+            statement.setInt(1, point.getId());
+            statement.setDouble(2, point.getXCoordinate());
+            statement.setDouble(3, point.getYCoordinate());
+            statement.executeUpdate();
+
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            if (generatedKeys.next()) {
```

Comment:
It may be useful to use the while loop here instead of the if loop to cover more scenarious

Reply:
changed